### PR TITLE
Refine on RSAKey id

### PR DESCRIPTION
**Pull request by student-pwr-262257**

### CODE REVIEW

File: todo_project_name/rsa.py [hunk 1]
```diff
@@ -6,21 +6,46 @@
...
+
+        self.key = key
+        self.modulus = modulus
+        self.id = id
+
+    def __repr__(self) -> str:
+        name = self.__class__.__name__
+        return f"{name}(key={self.key!r}, modulus={self.modulus!r}, id={self.id!r})"
+
+    def __eq__(self, other) -> bool:
```
Comment (Esgeriath) [inline todo_project_name/rsa.py:37]: Id is irrelevant?

File: todo_project_name/rsa.py [hunk 2]
```diff
@@ -72,11 +98,13 @@ def save_key(key: RSAKey, path: Path) -> Path:
 def read_key(path: Path, key_type: Type[RSAKeyVar]) -> RSAKeyVar:
     """Read RSA key from the file."""
     with path.open("r", encoding="utf8") as file:
+        # Skip the heading.
```
Comment (Esgeriath) [inline todo_project_name/rsa.py:101]: perhaps we could check if heading matches type of key we are supposed to be reading

File: todo_project_name/rsa.py [hunk 3]
```diff
@@ -61,7 +86,8 @@ def save_key(key: RSAKey, path: Path) -> Path:
     kind = key.__class__.__name__
     header = f"-----BEGIN {kind} KEY-----"
     footer = f"-----END {kind} KEY-----"
-    contents = "\n".join((header, str(key.key), str(key.modulus), footer))
+    # TODO: Consider serialization method allowing white space in `RSAKey.id`. Is it needed?
```
Comment (Esgeriath) [inline todo_project_name/rsa.py:89]: That probably will be resolved when  standard encoding will be implemented.

**Review by Esgeriath — APPROVED** [review]

File: todo_project_name/rsa.py (hunk 1 above)
Comment (student-pwr-262257) [reply todo_project_name/rsa.py:37]: To the comparison? I believe, yes. They are not the same object, obviously, but they allow for the same things in application. No matter which equal object you choose, you can encrypt or decrypt the same data.

File: todo_project_name/rsa.py (hunk 2 above)
Comment (student-pwr-262257) [reply todo_project_name/rsa.py:101]: Yes, I think that is a good idea. I haven't thought about it in the first implementation a few pushes ago, so I simply didn't change it this time (just added a comment). I would defer the change until we establish the format, if any. The change would be within the same function, so I don't think some “todo” comment is necessary.

File: todo_project_name/rsa.py (hunk 3 above)
Comment (student-pwr-262257) [reply todo_project_name/rsa.py:89]: Yes, I suppose. May this note serve as a reminder.

**Review by CarlGausss — COMMENTED** [review]
Excellent job, Jakub

**Pull request merged**